src/AWS/CRT/Options.php

```diff
@@ -5,15 +5,46 @@
  */
 namespace AWS\CRT;
 
+final class OptionValue {
+    private $value;
+    function __construct($value) {
+        $this->value = $value;
+    }
+
+    public function asObject() {
+        return $this->value;
+    }
+
+    public function asMixed() {
+        return $this->value;
+    }
+
+    public function asInt() {
+        return is_null($this->value) ? 0 : (int)$this->value;
```

Comment:
empty() may be a better thing to use here?  Empty will return true on more values, is null only checks if it's null.  I think this should work for the most part because falsy values will evaluate to zero, but what's the ideal behavior if say $this->value is "true".  That would evaluate to zero.

Reply:
I think that's reasonable-ish. If it contains the string `"true"` then that is not a number, and should return -1/0, whatever conversion says is the invalid value.